src/Core/Security/Infrastructure/Repository/DbReadAccessGroupRepository.php

```diff
@@ -1,65 +1,121 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* For more information : contact@centreon.com
+*
+*/
+
 declare(strict_types=1);
 
 namespace Core\Security\Infrastructure\Repository;
 
+use Centreon\Infrastructure\DatabaseConnection;
 use Centreon\Domain\Contact\Interfaces\ContactInterface;
-use Core\Security\Domain\AccessGroup\Model\AccessGroup;
+use Centreon\Domain\RequestParameters\RequestParameters;
+use Centreon\Infrastructure\Repository\AbstractRepositoryDRB;
 use Core\Security\Application\Repository\ReadAccessGroupRepositoryInterface;
-use Centreon\Infrastructure\DatabaseConnection;
+use Centreon\Infrastructure\RequestParameters\SqlRequestParametersTranslator;
 
 /**
  * Database repository for the access groups.
  *
  * @package Centreon\Infrastructure\Security
  */
-final class DbReadAccessGroupRepository implements ReadAccessGroupRepositoryInterface
+final class DbReadAccessGroupRepository extends AbstractRepositoryDRB implements ReadAccessGroupRepositoryInterface
 {
     /**
-     * @var DatabaseConnection
+     * @var SqlRequestParametersTranslator
+     */
+    private SqlRequestParametersTranslator $sqlRequestTranslator;
+
+    /**
+     * @param DatabaseConnection $db
+     * @param SqlRequestParametersTranslator $sqlRequestTranslator
      */
-    private $pdo;
+    public function __construct(DatabaseConnection $db, SqlRequestParametersTranslator $sqlRequestTranslator)
+    {
+        $this->db = $db;
+        $this->sqlRequestTranslator = $sqlRequestTranslator;
+        $this->sqlRequestTranslator
+            ->getRequestParameters()
+            ->setConcordanceStrictMode(RequestParameters::CONCORDANCE_MODE_STRICT);
+
+        $this->sqlRequestTranslator->setConcordanceArray([
+            'id' => 'acl_group_id',
+            'name' => 'acl_group_name'
+        ]);
+    }
 
     /**
-     * DbReadAccessGroupRepository constructor.
-     *
-     * @param DatabaseConnection $pdo
+     * @inheritDoc
      */
-    public function __construct(DatabaseConnection $pdo)
+    public function findAll(): array
     {
-        $this->pdo = $pdo;
+        $request = "SELECT SQL_CALC_FOUND_ROWS * FROM acl_groups";
+        $searchRequest = $this->sqlRequestTranslator->translateSearchParameterToSql();
+        $request .= $searchRequest !== null
+            ? $searchRequest . ' AND '
+            : ' WHERE ';
+
+        $request .= "acl_group_activate = '1'";
+
+        // Sort
+        $sortRequest = $this->sqlRequestTranslator->translateSortParameterToSql();
+        $request .= $sortRequest !== null ? $sortRequest : ' ORDER BY acl_group_id ASC';
+
+        // Pagination
+        $request .= $this->sqlRequestTranslator->translatePaginationToSql();
+
+        $statement = $this->db->prepare($request);
+
+        foreach ($this->sqlRequestTranslator->getSearchValues() as $key => $data) {
+            $type = key($data);
```

Comment:
are you sure about using of key function into the loop ?
https://www.php.net/manual/fr/function.key.php#114905